spring-cloud-gateway-server-mvc/src/main/java/org/springframework/cloud/gateway/server/mvc/filter/AfterFilterFunctions.java

```diff
@@ -160,6 +173,38 @@ public static BiFunction<ServerRequest, ServerResponse, ServerResponse> setStatu
 		};
 	}
 
+	public static BiFunction<ServerRequest, ServerResponse, ServerResponse> removeJsonAttributesResponseBody(
+			List<String> fieldList, boolean deleteRecursively) {
+		List<String> immutableFieldList = List.copyOf(fieldList);
+
+		return modifyResponseBody(String.class, String.class, APPLICATION_JSON_VALUE, (request, response, body) -> {
+			String responseBody = body;
+			if (APPLICATION_JSON.isCompatibleWith(response.headers().getContentType())) {
+				try {
+					JsonNode jsonBodyContent = OBJECT_MAPPER.readValue(responseBody, JsonNode.class);
+
+					removeJsonAttributes(jsonBodyContent, immutableFieldList, deleteRecursively);
+
+					responseBody = OBJECT_MAPPER.writeValueAsString(jsonBodyContent);
+				}
+				catch (JsonProcessingException exception) {
+					throw new IllegalStateException("Failed to process JSON of response body.", exception);
```

Comment:
What will happen if this exception is not handled in the configuration as you do in the test?

Reply:
@ryanjbaxter 

If the global exception is not handled as in the test configuration, the request will not be routed to a specific handler. 
Instead, the `DispatcherServlet` will forward the request to the `/error` path for default error handling.